### ISSUE TITLE
handshake verification 

### ISSUE THREAD

**Issue by josha33** [wifiphisher/wifiphisher#1067]
First of all thanks for the great tool.
I have problem with handshake verification.
I capture the handshake file .cap with airodump or .pcap with wireshark/tcpdump  and run wifiphisher.
when the user inputs passphrase it doesn't validated with the handshake file provided and it asks for passphrase again and again. this is what i get in the log file 

![wifiphisherlog](https://user-images.githubusercontent.com/47193323/52168778-84740980-2743-11e9-8058-eee45724be77.jpg)

I checked the handshake file with aircrack-ng and it contains the handshake and i can verify the right password manually with aircrack-ng. 
repeated this with handshake file captured with wireshark , tcpdump.

**Comment by josha33** [comment]
I can get  the passphrases submitted but as i said the passphrases not verified against handshake in handshake file.
![handshake1](https://user-images.githubusercontent.com/47193323/52169008-e124f380-2746-11e9-85f4-bc87b38399e1.jpg)

**Comment by sophron** [comment]
Thanks for opening this. We will have a look.

**Comment by anakin1028** [comment]
When you do the attck, do you also get all the four key frames? Can you post both the precapture handshake frames and the under attacking four frames in this thread?

**Comment by josha33** [comment]
What you mean by under attacking four frames? should i capture frames while running wifiphisher? the user will put passphrases with post method in fake page so there will not be any authentication key frames if i capture frames while running wifiphisher?

**Comment by anakin1028** [comment]
The steps will be:
1. Run wifiphisher with your previous commands.
2. Start the wireshark.
3. Let the victims connect to the rogueAP
4. Watch the wireshark to see if all four handshake frames are captured since our algorithm will need all four frames captured.

**Comment by anakin1028** [comment]
Sorry. I was wrong. Do You use WPA2 or WPA? and have you try any other passphrase? Can you provide a non-sensitive paraphrase which causes this problem to us to reproduce your problem?

**Comment by josha33** [comment]
I use WPA2. No I didn't change my routers password. Should i provide you a captured handshake file with simple password?

**Comment by anakin1028** [comment]
Yes that would be great if you can provide the pcap file to us:)

**Comment by josha33** [comment]
here's the cap file. 

[testme.zip](https://github.com/wifiphisher/wifiphisher/files/2872157/testme.zip)

passphrase is : testme33

**Comment by josha33** [comment]
what information do you need?

this is the fake page when i run wifiphisher : http://prnt.sc/mop1am
when the user inputs something i can get it :+1: http://prnt.sc/mop0c4
but if the user inputs the right passphrase it keeps asking for it (it can't validate passphrase with the handshake file provided).
if i cancel wifiphisher by esc : http://prnt.sc/mop6rd

i can verify passphrase by aircrack-ng : http://prnt.sc/mop7nt

**Comment by anakin1028** [comment]
I think the information is enough. Hopefully I will try to reproduce it in the following days. Thanks :)

**Comment by pr-lad** [comment]
@josha33  Seems the template you are using is not calling the extension to verify credentials.

I know the code below must be put inside the template, however I have no idea where it would go.

Can anyone with more knowledge on this topic suggest a solution?

@anakin1028  any thoughts?


```
var data =
{
    "psk_verify": input.value // captured creds
};
var dataToSend = JSON.stringify(data);
// post the data
$.ajax(
    {
        url: '/backend/',
        type: 'POST',
        data: dataToSend,

        success: function (jsonResponse) {
            var objresponse = JSON.parse(jsonResponse);
            var verify_status = objresponse['psk_verify']
            if (verify_status == 'success') {
               // Print Success Message
            } else if (verify_status == 'fail') {
               // Credentials are invalid. Ask the victim user again.
            }
       }
    }
 );
```

**Comment by minanagehsalalma** [comment]
> Can anyone with more knowledge on this topic suggest a solution?

@sophron see ! The same problem here and it have nothing to do with script kiddies !
